### PR TITLE
ci: remove old compliance tests & enable new compliance tests on Windows

### PR DESCRIPTION
**Pull request by petebacondarwin**

### CODE REVIEW

**Review by AndrewKushnir — APPROVED** [review]
Thanks for doing this cleanup @petebacondarwin 👍

File: packages/compiler-cli/test/compliance/partial/partial_compliance_goldens.bzl [hunk 1]
```diff
@@ -48,10 +48,6 @@ def partial_compliance_golden(filePath):
         testonly = True,
         stdout = "%s/this_file_should_not_be_committed" % path,
         link_workspace_root = True,
-        tags = [
-            # TODO(josephperrott): Begin running these tests on windows after updating to rules_nodejs 3.0
-            "no-windows",
```
Comment (josephperrott) [inline packages/compiler-cli/test/compliance/partial/partial_compliance_goldens.bzl:53]: As these are the only `no-windows` usages, we can also remove the usage from the `.circleci/config.yaml` file as well

**Review by josephperrott — CHANGES_REQUESTED** [review]

File: packages/compiler-cli/test/compliance/partial/partial_compliance_goldens.bzl (hunk 1 above)
Comment (petebacondarwin) [reply packages/compiler-cli/test/compliance/partial/partial_compliance_goldens.bzl:53]: Done.

**Review by josephperrott — APPROVED** [review]
LGTM

**Comment by zarend** [comment]
Hi @petebacondarwin . This was merged into `master` but not merged into the patch branch due to a merge conflict.

**Pull request closed without merging**